File: Data Structures & Algorithms/validate-parentheses/submission-6.py
opens = {'}': '{', ']': '[', ')': '('}
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if i in ['[', '(', '{']:
                stack.append(i)
                continue
            
            if len(stack) == 0:
                return False
            x = stack.pop()
            if x != opens[i]:
                return False
        return not stack

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-6.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    operators = {
        '+': (a, b) => a + b,
        '-': (a, b) => a - b,
        '*': (a, b) => a * b,
        '/': (a, b) => a / b >= 0 ? Math.floor(a / b) : Math.ceil(a / b)
    }
    evalRPN(tokens) {
        const stack = []

        for (const token of tokens) {
            if (this.operators[token]) {
                const second = stack.pop()
                const first = stack.pop()
                const result = this.operators[token](first, second)
                console.log(result)
                stack.push(result)
            } else {
                stack.push(Number(token))
            }
        }
        return stack.pop()
    }
}
